python/lsst/pipe/tasks/postprocess.py

```diff
@@ -330,75 +312,347 @@ def run(self, catalog, ccdVisitId=None):
+        externalPhotoCalibGlobalCatalog : `lsst.afw.table.ExposureCatalog`, optional
+            Exposure catalog with external photoCalib to be applied per config
+        externalPhotoCalibTractCatalog : `lsst.afw.table.ExposureCatalog`, optional
+
 
         Returns
         -------
-        newCat:  `afwTable.SourceCatalog`
-            Source Catalog with requested local calib columns
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Exposure with adjusted calibrations.
         """
-        mapper = afwTable.SchemaMapper(catalog.schema)
-        measureConfig = SingleFrameMeasurementTask.ConfigClass()
-        measureConfig.doReplaceWithNoise = False
+        if self.config.useGlobalExternalSkyWcs:
+            externalSkyWcsCatalog = externalSkyWcsGlobalCatalog
+            self.log.info('Applying global SkyWcs')
+        else:
+            # get closest overlapping tract
+            if externalSkyWcsTractCatalog:
+                inputRef = getattr(inputRefs, 'externalSkyWcsTractCatalog')
+                tracts = [ref.dataId['tract'] for ref in inputRef]
+                if len(tracts) == 1:
+                    ind = 0
+                    self.log.info('Applying tract-level SkyWcs from tract %s', tracts[ind])
+                else:
+                    ind = self.getClosestTract(tracts, skyMap,
+                                               exposure.getBBox(), exposure.getWcs())
+                    self.log.info('Multiple overlapping externalSkyWcsTractCatalogs found (%s). '
+                                  'Applying closest to detector center: tract=%s', str(tracts), tracts[ind])
+            externalSkyWcsCatalog = externalSkyWcsTractCatalog[ind]
+
+        if self.config.useGlobalExternalPhotoCalib:
+            externalPhotoCalibCatalog = externalPhotoCalibGlobalCatalog
+            self.log.info('Applying global PhotoCalib')
+        else:
+            if externalPhotoCalibTractCatalog:
+                inputRef = getattr(inputRefs, 'externalPhotoCalibTractCatalog')
+                tracts = [ref.dataId['tract'] for ref in inputRef]
+                if len(tracts) == 1:
+                    ind = 0
+                    self.log.info('Applying tract-level PhotoCalib from tract %s', tracts[ind])
+                else:
+                    ind = self.getClosestTract(tracts, skyMap,
+                                               exposure.getBBox(), exposure.getWcs())
+                    self.log.info('Multiple overlapping externalPhotoCalibTractCatalogs found (%s). '
+                                  'Applying closest to detector center: tract=%s', str(tracts), tracts[ind])
 
-        # Just need the WCS or the PhotoCalib attached to an exposue
-        exposure = dataRef.get('calexp_sub',
-                               bbox=lsst.geom.Box2I(lsst.geom.Point2I(0, 0), lsst.geom.Point2I(0, 0)))
+            externalPhotoCalibCatalog = externalPhotoCalibTractCatalog[ind]
```

Comment:
Should this line be inside the `if externalPhotoCalibTractCatalog:` statement? Looks like it is not.

Reply:
I totally see why this was confusing. Making clearer.